Todo/TasksViewController.swift

```diff
@@ -0,0 +1,353 @@
+            
+            return [delete, update]
+    }
+    
+    // MARK: - UISearchController
+    
+    func updateSearchResultsForSearchController(searchController: UISearchController) {
+        let text = searchController.searchBar.text ?? ""
+        let active = !text.isEmpty
+        tasksLiveQuery.postFilter = active ? NSPredicate(format: "key2 CONTAINS[cd] %@", text) : nil
+        tasksLiveQuery.queryOptionsChanged()
+    }
+    
+    // MARK: - UIImagePickerControllerDelegate
+    
+    func imagePickerController(picker: UIImagePickerController,
+        didFinishPickingImage image: UIImage, editingInfo: [String : AnyObject]?) {
+            if let task = taskForImage {
+                updateTask(task, withImage: image)
+                self.taskForImage = nil
+            }
+            picker.presentingViewController?.dismissViewControllerAnimated(true, completion: nil)
+    }
+    
+    // MARK: - Action
+    
+    func addAction(sender: AnyObject) {
+        Ui.showTextInputDialog(
+            onController: self,
+            withTitle: "New Task",
+            withMessage: nil,
+            withTextFieldConfig: { (textField) -> Void in
+                textField.placeholder = "Task"
+                textField.autocapitalizationType = .Sentences
+            },
+            onOk: { (task) -> Void in
+                self.createTask(task)
+            }
+        )
+    }
+    
+    // MARK: - Database
+    
+    func setupViewAndQuery() {
+        let view = database.viewNamed("tasksByCreatedAt")
+        if view.mapBlock == nil {
+            view.setMapBlock({ (doc, emit) -> Void in
+                if let type: String = doc["type"] as? String where type == "task" {
+                    if let list = doc["taskList"] as? [String: AnyObject] {
+                        if let listId = list["id"], createdAt = doc["createdAt"], task = doc["task"] {
+                            emit([listId, createdAt, task], nil)
+                        }
+                    }
+                }
+            }, version: "1.0")
+        }
+        
+        tasksLiveQuery = view.createQuery().asLiveQuery()
+        tasksLiveQuery.startKey = [taskList.documentID]
+        tasksLiveQuery.endKey = [taskList.documentID]
+        tasksLiveQuery.prefixMatchLevel = 1
+        tasksLiveQuery.descending = true
+        
+        tasksLiveQuery.addObserver(self, forKeyPath: "rows", options: .New, context: nil)
+        tasksLiveQuery.start()
+    }
+    
+    func reloadTasks(queryEnum: CBLQueryEnumerator?) {
+        taskRows = queryEnum?.allObjects as? [CBLQueryRow] ?? nil
+        tableView.reloadData()
+    }
+    
+    func createTask(task: String) {
+        let taskListInfo: Dictionary<String, AnyObject> = [
+            "id": taskList.documentID,
+            "owner": taskList["owner"]!
+        ]
+        
+        let properties: Dictionary<String, AnyObject> = [
```

Comment:
When the value type are mixed, without declaring the type, I will need to cast it when calling `doc.putProperties`. BUT it also has a weird additional compilation error that `Errors thrown from here are not handled because the enclosing catch is not exhaustive`. To solve that error, I need to add an extra catch. I couldn't figure out why I don't need to do that when I explicitly declared the type.

```
do {
    try doc.putProperties(properties as! [String : AnyObject])
} catch let error as NSError {
    Ui.showMessageDialog(onController: self, withTitle: "Error",
        withMessage: "Couldn't save task", withError: error)
} catch { 
}
```